packages/flutter/lib/src/widgets/actions.dart

```diff
@@ -383,11 +385,25 @@ class ActionDispatcher with Diagnosticable {
   /// Returns the object returned from [Action.invoke] if the action was
   /// successfully invoked, and null if the action is not enabled. May also
   /// return null if [Action.invoke] returns null.
-  Object? invokeAction(covariant Action<Intent> action, covariant Intent intent, [BuildContext? context]) {
+  ///
+  /// If `isEnabled` is supplied, and true, then the action will be invoked
```

Comment:
BTW, I'm not terribly happy with this API. Since `isEnabled` could do arbitrary work to determine if it should be enabled or not, I didn't want to call it once to determine if it was enabled to decide if I should handle it or not, and then again here when it was invoked.

If you have any better ideas, I'd love to hear them.

Reply:
Given that most users of this interface will probably be calling `Shortcuts.handleKeypress` to invoke things, it's probably not that breaking. I'm willing to give it a try and see if anyone complains. I doubt it would break a test.